src/main/java/com/infomaniak/lib/core/auth/CredentialManager.kt

```diff
@@ -17,15 +17,95 @@
  */
 package com.infomaniak.lib.core.auth
 
+import androidx.collection.ArrayMap
 import androidx.lifecycle.LiveData
+import com.facebook.stetho.okhttp3.StethoInterceptor
+import com.infomaniak.lib.core.BuildConfig
 import com.infomaniak.lib.core.models.user.User
+import com.infomaniak.lib.core.room.UserDatabase
+import com.infomaniak.lib.login.ApiToken
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import okhttp3.OkHttpClient
+import java.util.concurrent.TimeUnit
 
 /**
  * CredentialManager interface : Implement the essential methods to get Users and their Credentials to pass
  */
-interface CredentialManager {
+abstract class CredentialManager {
+
+    abstract var userDatabase: UserDatabase
+    abstract var currentUserId: Int
+    abstract var currentUser: User?
+
     /**
      * Get users, and their informations / tokens in a JSON format
      */
-    fun getAllUsers(): LiveData<List<User>>
-}
\ No newline at end of file
+    abstract fun getAllUsers(): LiveData<List<User>>
```

Comment:
It would also be nice to define the body of this method here, as it is always the same one that is shared.
As well as the `getAllUserCount` method